### PR TITLE
Add multi-comments

### PR DESCRIPTION
**Pull request by jiangtj**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [x] [Docs](https://github.com/theme-next/theme-next.org/tree/source/source/docs) in [NexT website](https://theme-next.org/docs/) have been added / updated (for features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?
Not support multi-comments.

Issue resolved: https://github.com/iissnan/hexo-theme-next/issues/1604

## What is the new behavior?
Add multi-comments support.

- https://jiangtj.gitlab.io/next-test/gemini/post/hello-world/

### How to use?
In NexT `_config.yml`:
```yml
# Multiple Comment System Support
comments:
  # Available values: tabs | buttons
  style: tabs
  # Choose a comment system to be displayed by default
  # Available values: changyan | disqus | disqusjs | facebook_comments_plugin | gitalk | livere | valine | vkontakte
  active:
  # Setting `true` means remembering the comment system selected by the visitor
  storage: true
  # Modify icons and texts for any style, here are some examples
  nav:
    #disqus: disqus
    #facebook_comments_plugin: <i class="fa fa-facebook-official" aria-hidden="true"></i> facebook
    #gitalk: Load Gitalk
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by jiangtj** [comment]
Disqus and disqusjs cannot be enabled at the same time

**Comment by 1v9** [comment]
No problem, they are same 🐱‍👤

**Comment by jiangtj** [comment]
image

post meta ....

File: scripts/filters/comment/default-config.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* global hexo */
+
+'use strict';
+
+const path = require('path');
+
+hexo.extend.filter.register('theme_inject', function(injects) {
```
Comment (1v9) [inline scripts/filters/comment/default-config.js]: arrow function

**Comment by jiangtj** [comment]
> Is there reason to use tabs (tag) instead of buttons? For better and instantly block with comments alignment?

image
See this photo.
I prefer buttons, maybe we can provide multi style.

**Comment by sli1989** [comment]
post meta is a big problem. for some comments systems, there have integrated with `1 comments`. this is also considered with post metas in comments with muilti-langanges.

**Comment by jiangtj** [comment]
> post meta is a big problem. for some comments systems, there have integrated with `1 comments`. this is also considered with post metas in comments with muilti-langanges.

I have considered   `💬 gitalk: 1| 💬 valine: 1`.  This may cause `post-meta` be longer.

Another issues is that if we hide text desc (me), It will be  `💬 1| 💬 1`. 
Most comment systems don't have a unique icon Orz....So I prefer to hide all comment's `post-meta`.

**Comment by ivan-nginx** [comment]
1. Maybe add option: tabs (default) or buttons. They auto-wrapped.
2. Values in post-meta must be auto-wrapped too. Must be fine like below:

> > post meta is a big problem. for some comments systems, there have integrated with `1 comments`. this is also considered with post metas in comments with muilti-langanges.
> 
> I have considered `💬 gitalk: 1| 💬 valine: 1`. This may cause `post-meta` be longer.
> 
> Another issues is that if we hide text desc (me), It will be `💬 1| 💬 1`.
> Most comment systems don't have a unique icon Orz....So I prefer to hide all comment's `post-meta`.

Only one thing here will be good: add titles with comment system names like titles in date-created & date-updated:

image

**Comment by jiangtj** [comment]
> Only one thing here will be good: add titles with comment system names like titles in date-created & date-updated:

Added

**Comment by jiangtj** [comment]
Approved?

File: _config.yml [hunk 2]
```diff
@@ -609,6 +609,21 @@ reading_progress:
...
 # ---------------------------------------------------------------
+# Comments system setting
+# Types: changyan | disqus | disqusjs | facebook-comments-plugin | gitalk | livere | valine | vkontakte-comments
+comments:
+  # You can choose `buttons` or `tabs`
+  style: tabs
+  # If you enable mutli comments, you can active one comment type, it will display as default when page loaded.
+  active:
+  # If true, storage user selected comment, and active this when page loaded.
+  # Only support style is `buttons`.
```
Comment (1v9) [inline _config.yml]: supported

File: layout/_partials/comments.swig [hunk 3]
```diff
@@ -2,8 +2,60 @@
...
+              commentButton.forEach(rmActive => rmActive.classList.remove('active'));
+              item.classList.add('active');
+              document.querySelectorAll('.comment-position').forEach(rmActive => rmActive.classList.remove('active'));
+              document.querySelector(`.comment-position.${commentClass}`).classList.add('active');
+              {%- if theme.comments.storage %}
+                localStorage.setItem('comments_active', commentClass);
+              {%- endif %}
+            })
+          })
+          let activeClass = '{{ theme.comments.activeClass }}';
```
Comment (stevenjoezhang) [inline layout/_partials/comments.swig]: `theme.comments.activeClass` is not defined

Comment (jiangtj) [reply layout/_partials/comments.swig]: image
In here, set `activeClass`. Make sure class name is `facebook-comments-plugin`, and config key name is `facebook_comments_plugin`

**Comment by ivan-nginx** [comment]
Docs on website needed?

**Comment by jiangtj** [comment]
> Docs on website needed?

Need

**Comment by sli1989** [comment]
seems there is a spare pace after the number of the comment?

**Comment by jiangtj** [comment]
> seems there is a spare pace after the number of the comment?

A spare pace? where? Can you cut a picture?

**Comment by sli1989** [comment]
image

**Comment by jiangtj** [comment]
image
Not a space, the same width is OK？

**Comment by sli1989** [comment]
u see? the space after `1 comment` & `0` is differrent.

**Comment by jiangtj** [comment]
image
This space? 😂 This problem existed before, and there are some space between </span></a>

**Pull request merged**